src/instructlab/model/evaluate.py

```diff
@@ -60,6 +60,16 @@ def get_evaluator(
     otherwise raises an exception for the missing values
     """
 
+    # ensure user is passing full safetensors if they specify a local directory
+    # TODO: also allow GGUF once the following is resolved: https://github.com/instructlab/eval/issues/50
+    if model is not None and os.path.isdir(model):
```

Comment:
So if someone has a path to a gguf file as opposed to its directory, this will be false. Is that what you want?

Reply:
Hmm, yeah I noted this in the bug https://github.com/instructlab/instructlab/issues/1792#issuecomment-2239811057 - I'm not sure if we _never_ are going to be able to use a specific file or if we're always going to require either a directory or HF repo. @alimaredia @alinaryan do either of you know?

Reply:
If files are verbotten, I think it'll be worth adding the following to this PR:
1. Additional check to ensure the we are not passing a local file
2. Additional unit test to ensure passing a file leads to an expected excepton

Reply:
The `isdir` check is probably redundant btw, since I see the following inside `is_model_safetensors`:

```
    try:
        files = list(model_path.iterdir())
    except (FileNotFoundError, NotADirectoryError, PermissionError) as e:
        logger.debug("Failed to read directory: %s", e)
        return False
```

Note the `NotADirectoryError` is already handled there.

Reply:
The reason I added that there is because `model` in this case gets passed as a string, rather than a `pathlib.Path` type, so when you try to call `iterdir()` it throws an unhandled exception. 

This gets into the territory of what @JamesKunstle is doing in https://github.com/instructlab/instructlab/pull/1791 :smile:

Reply:
That it's passed into is_model_safetensors as a string is a bug of the caller, and should be fixed in the caller. (Commented in the linked PR too.)

Reply:
That's what I'm doing here no? Thus the `isdir` check (though as @russellb points out we may want to abstract this to a directory or file path)

Reply:
@danmcp can you confirm if there's any case where a singular file would be a valid input, or do we always need to be doing a local dir/HF repo?

Reply:
if the former, I can change to `os.path.exists()`

Reply:
Side note re: `if model is not None` part of the check: I don't think long term we should accept a None model here. See: https://github.com/instructlab/instructlab/pull/1810 At least as of right now, the code below in launch_server will raise TypeError if None ever sneaks past get_evaluator. I think it's better to just enforce it via click.option unless there's a use case for a missing --model?

Reply:
Thanks for opening that @booxter! AFAIK we don't have a case for a missing `--model` - could someone else from the Eval Squad confirm?

Reply:
The `if model is not None` was removed in my latest patchset, so shouldn't be any conflict there

Reply:
From this discussion, it's not clear if `model` can ever be a path to a specific file. If so, this still seems insufficient.

Reply:
I've done some manual testing around this and made the following findings wrt the model arguments in additional to the discussion above:
1. MMLU/MMLUBranch require safetensors, either local or on HF, so this check should be sufficient for that
2. MTBench/MTBenchBranch must have a local path, either to a directory with safetensors or a GGUF file
I am going to update this PR to account for both, changing the original scope. Will update the PR title and commit messages to account for this.